Implemented remote deployment

rogeralsing opened the pull request:
This PR implements the remote deploy functionality.
The main aspect to review is how to group the Remote API.
Currently we have `RemotingSystem` as the API surface for all actions.
First, the package has been renamed from Remoting to Remote, so the name makes no sense anymore.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AsynkronIT/protoactor-dotnet?pullRequest=88) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AsynkronIT/protoactor-dotnet?pullRequest=88) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AsynkronIT/protoactor-dotnet?pullRequest=88) it.</sub>

cpx86 commented:
Looks nice and clean to me! Two small things:
1. Agree on the naming. We should at least rename it to RemoteSystem to make it more consistent. Not sure about the System suffix either (sounds a bit like actor system, which we don't have) but unless we think of something else that's fine.
2. Would be nice to have an example showing how this works! :)

rogeralsing commented:
Added RemoteActivate example and renamed RemotingSystem to just Remote.
Potential issue `Proto.Remote.Remote` but it does look OK from the current code.

cpx86 commented:
Awesome!
